### DISCUSSION TITLE
CurrentUser incorrectly resolves empty User instance

### DISCUSSION THREAD

**Discussion by anned20** [laravel/framework#53274] [Ideas]
### Laravel Version

11.22.0

### PHP Version

8.2.16

### Description

We are using the new `ContextualAttribute` to get the `CurrentUser` in our controller methods. This works fine, until there is no user in the guard.

The function `resolve` in the attribute correctly returns `null`, but the `Container` then resolves the parameter to a new `User` instance if there is a type-hint for a `User` on the parameter. This even happens when the parameter type is indicated as optional.

### Steps To Reproduce

```php
class SomeController
{
    public function someMethod(
        #[CurrentUser]
        ?User $user
    ): void {
        dd($user);
    }
}
```

When executing this code without being logged in (having no available user), the `$user` will contain an empty model.

**Comment by EranNL** [comment]
This could also potentially "break" some edge cases in the new features from this PR: https://github.com/laravel/framework/pull/53080
When you have an optional parameter in your URL, I guess you would not want an empty model when trying to resolve it, it would be better to resolve to `null` in my opinion.

**Comment by alen-androsevic** [comment]
Any fixes?
